instrumentation/httptrace/httptrace_test.go

```diff
@@ -415,3 +415,239 @@ func TestStartRequestSpanMergedBaggage(t *testing.T) {
+
+	// Start the request span, which will extract header baggage and merge it with the context's baggage.
+	span, _, _ := StartRequestSpan(req)
+	m := span.AsMap()
+
+	v, ok := m["baggage.account.id"]
+	assert.True(t, ok, "baggage.account.id not found in span")
+	assert.Equal(t, "456", v, "should contain account.id value")
+
+	v, ok = m["baggage.session.id"]
+	assert.True(t, ok, "baggage.session.id not found in span")
+	assert.Equal(t, "789", v, "should contain session.id value")
+
+	_, ok = m["baggage.header_key"]
+	assert.False(t, ok, "baggage.header_key should not be included in span tags")
+
+	_, ok = m["baggage.user.id"]
+	assert.False(t, ok, "baggage.user.id should not be included in span tags")
+
+	span.Finish()
+}
+
+func TestBaggageSpanTagsWildcard(t *testing.T) {
+	os.Setenv("DD_TRACE_BAGGAGE_TAG_KEYS", "*")
+	ResetCfg()
+	tracer.Start()
+	defer tracer.Stop()
+
+	// Create an HTTP request with that context.
+	req := httptest.NewRequest(http.MethodGet, "/somePath", nil).WithContext(context.Background())
+
+	// Set the baggage header with additional baggage items.
+	req.Header.Set("baggage", "user.id=abcd,account.id=456,session.id=789,color=blue,foo=bar")
+
+	// Start the request span, which will extract header baggage and merge it with the context's baggage.
+	span, _, _ := StartRequestSpan(req)
+	m := span.AsMap()
+	// Assert that the expected baggage keys are present with the correct values.
+	v, _ := m["baggage.account.id"]
+	assert.True(t, v != "", "baggage.account.id not found in span")
+	assert.Equal(t, "456", v, "should contain account.id value")
+
+	v, _ = m["baggage.user.id"]
+	assert.True(t, v != "", "baggage.user.id not found in span")
+	assert.Equal(t, "abcd", v, "should contain user.id value")
+
+	v, _ = m["baggage.session.id"]
+	assert.True(t, v != "", "baggage.session.id not found in span")
+	assert.Equal(t, "789", v, "should contain session.id value")
+
+	v, _ = m["baggage.color"]
+	assert.True(t, v != "", "baggage.color not found in span")
+	assert.Equal(t, "blue", v, "should contain color value")
+
+	v, _ = m["baggage.foo"]
+	assert.True(t, v != "", "baggage.foo not found in span")
+	assert.Equal(t, "bar", v, "should contain foo value")
+
+	span.Finish()
+}
+
+func TestBaggageSpanTagsDisabled(t *testing.T) {
+	os.Setenv("DD_TRACE_BAGGAGE_TAG_KEYS", "")
+	ResetCfg()
+	tracer.Start()
+	defer tracer.Stop()
+
+	// Create an HTTP request with that context.
+	req := httptest.NewRequest(http.MethodGet, "/somePath", nil).WithContext(context.Background())
+
+	// Set the baggage header with additional baggage items.
+	req.Header.Set("baggage", "user.id=abcd,account.id=456,session.id=789,color=blue")
+
+	// Start the request span, which will extract header baggage and merge it with the context's baggage.
+	span, _, _ := StartRequestSpan(req)
+	m := span.AsMap()
+	// no values should be present in span tags
+	if _, ok := m["baggage.account.id"]; ok {
```

Comment:
nit: We can use https://pkg.go.dev/github.com/stretchr/testify/assert#Contains for these assertions.

Reply:
➕